hera_sim/simulate.py

```diff
@@ -31,226 +39,386 @@ def new_func(*args, **kwargs):
     return new_func
 
 
-# FIXME: some of the code in here is pretty brittle and breaks (sometimes silently)
-# if not used carefully. This definitely needs a review and cleanup.
 class Simulator:
     """Class for managing a simulation.
 
     """
 
-    def __init__(self, data=None, defaults_config=None, **kwargs):
-        """Initialize a Simulator object.
-
-        Idea: Make Simulator object have three major components:
-            sim.data -> UVData object for storing the "measured" data
-                Also keep track of most metadata here
-            sim.defaults -> Defaults object
-
+    def __init__(
+        self,
+        *,
+        data: Optional[Union[str, UVData]] = None,
+        defaults_config: Optional[Union[str, Dict]] = None,
+        redundancy_tol: float = 1.0,
+        **kwargs,
+    ):
+        """Simulate visibilities and instrumental effects for an entire array.
+
+        Parameters
+        ----------
+        data
+            ``pyuvdata.UVData`` object to use for the simulation or path to a
+            UVData-supported file.
+        defaults_config
+            Path to defaults configuraiton, seasonal keyword, or configuration
+            dictionary for setting default simulation parameters. See tutorial
+            on setting defaults for further information.
+        redundancy_tol
+            Position tolerance for finding redundant groups, in meters. Default is
+            1 meter.
+        kwargs
+            Parameters to use for initializing UVData object if none is provided.
+            If ``data`` is a file path, then these parameters are used when reading
+            the file. Otherwise, the parameters are used in creating a ``UVData``
+            object using ``io.empty_uvdata``.
```

Comment:
use `:func:~io.empty_uvdata`

Reply:
Addressed this and all other occurrences of this issue.